### ISSUE TITLE
fingerprint support for oneplus 3?

### ISSUE THREAD

**Issue by kissstudio** [ubports/ubuntu-touch#1315]
I found fingerprint unlock on meizu is supported, is it possible take it to oneplus3?

**Comment by Flohack74** [comment]
Well its not really working in the new xenial image we use. The component used by Canonical has some closed source stuff which was only working for vivid I think. See also here: https://github.com/ubports/ubuntu-touch/issues/396 - So I would consider this issue as being blocked by 396

**Comment by Vince1171** [comment]
To support fingerprint scanner for op3 we "just" have to implement [biometryd](https://biometryd.readthedocs.io/en/latest/) for the op3. 
The problem is: No one know how to do that.
As Flohack said, the only known implementation is for the meizu pro5 but we don't have the source for it. But it's closed source. So we have no idea how it has been done...

**Comment by Flohack74** [comment]
Lets reverse-engineer it ;)

**Comment by kissstudio** [comment]
@Flohack74 well, good idea :)

**Comment by dobey** [comment]
What one _should_ do instead, is to implement a [plug-in](https://github.com/ubports/biometryd/blob/xenial/doc/extending_biometryd.md) for biometryd, which instead of being device-specific (as the proprietary pro 5 blob is), uses libhybris to access the biometry interface through the Android HAL. This would then be generic and should work on more devices.

**Comment by Flohack74** [comment]
Maybe we can borrow from Sailfish. They got something that might work. Our plugin would just need to have the code from fpd-slave probably:

`sailfish-fpd <> sailfish-fpd-slave <> libgbinder <> Android BSP HAL: android.hardware.fingerprint`

**Comment by kissstudio** [comment]
you mean the sailfish-fpd-slave is a bridge between   
Android service and uplevel os， so can use the code build one middleware to translate api calls between them？

**Comment by kissstudio** [comment]
will be a good idea if Sailfish support the op3 fingerprint.

**Comment by Flohack74** [comment]
Yes I mean that. But its unsure if we can use it in the same way, someone needs to investigate.

**Comment by Flohack74** [comment]
@dobey biometryd is not installed at all in our rootfs, should we prepare this to be able to play with it a bit better?

**Comment by dobey** [comment]
> @dobey biometryd is not installed at all in our rootfs, should we prepare this to be able to play with it a bit better?

What? Of course biometryd is installed in the rootfs. I'm not sure why you would think it isn't, as system-settings requires it.

**Comment by Flohack74** [comment]
Hmm sorry I thought its an overlay, so I only looked into ubports-meta lists...

**Comment by MoemUT** [comment]
If this ever happens, I will buy the person who makes it happen a beer! 🍻
